src/core/config/transformer/subgraph.rs

```diff
@@ -39,22 +39,24 @@ impl Transform for Subgraph {
             // if federation is disabled don't process the config
             return Valid::succeed(config);
         }
-
+        let new_config = config.clone();
```

Comment:
if we just want to check the availability of types, then don't clone the entire config just clone the types map.